src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -64,6 +71,12 @@ public sealed class Arguments : TransformInputBase
 
             [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "Name of the output column.", SortOrder = 2)]
             public string[] OutputColumns;
+
+            [Argument(ArgumentType.AtMostOnce | ArgumentType.Required, HelpText = "GPU device id to run on (e.g. 0,1,..). -1 runs on CPU. Requires CUDA 10.0.", SortOrder = 3)]
+            public int GpuDeviceId = -1;
```

Comment:
>public int GpuDeviceId = -1; [](start = 12, length = 28)

I suggest we use nullable<int> (int?) instead of int with a special -1 value. Int with special -1 as null, is a C/C++ usage because there's no built-in nullable type in C/C++. In C#, that type is supported, so we should use it. That way we don't have this magic -1 value floating all over the code. I suggest we change all C# methods that take GpuDeviceId to use int? instead of int. The value should be changed to int when OnnxRuntime is called at the very end.